src/core/job.c

```diff
@@ -778,7 +778,7 @@ static void job_log_status_message(Unit *u, JobType t, JobResult result) {
 
         /* The description might be longer than the buffer, but that's OK, we'll just truncate it here */
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        xsprintf(buf, format, unit_description(u));
+        snprintf(buf, sizeof(buf), format, unit_description(u));
```

Comment:
Thank you. Interestingly, the bug was once fixed in https://github.com/systemd/systemd/commit/07ecca0dc9d2d8f3b3abd73ab32f254f339fd903 with a follow-up in https://github.com/systemd/systemd/commit/e68eedbbdc98fa13449756b7fee3bed689d76493. Then it was all reintroduced in https://github.com/systemd/systemd/pull/7454. Would it make sense to add a test?
 I would also say that `xsprintf.cocci` should be removed or at least contain a comment explaining why it is not a good idea to apply it to some files in the tree.

Reply:
@evverx but that more or less applies to all of the coccinelle scripts.

I'm pretty sure we should add comments to the code locations where cocinelle would do the wrong thing. And in fact this location actually contained a comment about this, but not one that is clear enough I guess. Hence I think we should change this patch in two ways:

1. Explicitly cast the result of snprintf() to void, to tell any static checkers that we knowingly ignore the truncation issues here.
2. Extend the comment two lines earlier to explicitly say: "note that we use snprintf() rather than xsprintf() on purpose here: we are fine with truncation and don't consider that an error."

@iaguis any chance you can rework the patch like that?

Reply:
Sure, I'll do that when I get back to my keyboard

Reply:
Done.

Reply:
@iaguis, thank you.

@poettering, I agree that clear comments are better than unclear ones, but I doubt that any comments, however clear, can prevent regressions like this. I opened https://github.com/systemd/systemd/pull/8210, which should make it much harder to blindly apply the patch. I still think that `xsprintf.cocci` should be removed if this happens again, though.